tests/unit/lib/test_file.py

```diff
@@ -323,3 +326,50 @@ def test_read_text(tmp_path, catalog):
     file = File(path=file_name, source=f"file://{tmp_path}")
     file._set_stream(catalog, True)
     assert file.read_text() == data
+
+
+def test_file_resolve():
+    mock_catalog = MagicMock(spec=Catalog)
+    mock_client = MagicMock()
+    mock_catalog.get_client.return_value = mock_client
+
+    file = File(
+        source="s3://test-bucket",
+        path="test/file.txt",
+        size=0,
+        etag="",
+        version="",
+        is_latest=True,
+        last_modified=TIME_ZERO,
+    )
+    file._catalog = mock_catalog
+
+    mock_client.info.return_value = {
+        "size": 1024,
+        "etag": "abc123",
+        "version": "v1",
+        "is_latest": True,
+        "last_modified": datetime(2023, 1, 1, 12, 0, 0),
+    }
+
+    resolved_file = file.resolve()
+
+    assert resolved_file.size == 1024
+    assert resolved_file.etag == "abc123"
+    assert resolved_file.version == "v1"
+    assert resolved_file.is_latest
+    assert resolved_file.last_modified == datetime(2023, 1, 1, 12, 0, 0)
+
+    mock_client.info.side_effect = Exception("File not found")
```

Comment:
I'd suggest extracting this to a separate test. The same for the missing catalog.